dWorldServer/WorldServer.cpp

```diff
@@ -202,19 +202,21 @@ int main(int argc, char** argv) {
 	//Find out the master's IP:
 	std::string masterIP = "localhost";
 	uint32_t masterPort = 1000;
+	std::string masterPassword;
 	auto masterInfo = Database::Get()->GetMasterInfo();
 
 	if (masterInfo) {
 		masterIP = masterInfo->ip;
 		masterPort = masterInfo->port;
+		masterPassword = masterInfo->password;
 	}
 
 	UserManager::Instance()->Initialize();
 
 	const bool dontGenerateDCF = GeneralUtils::TryParse<bool>(Game::config->GetValue("dont_generate_dcf")).value_or(false);
 	Game::chatFilter = new dChatFilter(Game::assetManager->GetResPath().string() + "/chatplus_en_us", dontGenerateDCF);
 
-	Game::server = new dServer(masterIP, ourPort, instanceID, maxClients, false, true, Game::logger, masterIP, masterPort, ServerType::World, Game::config, &Game::lastSignal, zoneID);
+	Game::server = new dServer(masterIP, ourPort, instanceID, maxClients, false, true, Game::logger, masterIP, masterPort, ServerType::World, Game::config, &Game::lastSignal, masterPassword, zoneID);
```

Comment:
If master fails and restarts for some reason, generates a new pass, should we try to fetch the new pass in the main loop while disconnected?

Or is a disconnected master always a reason to kill all world servers

Reply:
We have infrasturcture setup for a failover to a new master server?

Reply:
> Or is a disconnected master always a reason to kill all world servers

I would think that would be the case

Reply:
> We have infrasturcture setup for a failover to a new master server?

I know people have put master server in a systemd service if that's what you mean

Reply:
by default systemd kills child processes if the main pid dies, instead of orphaning them
(from what I understand)